### ISSUE TITLE
[question]why validation after each epoch is divided into batches?

### ISSUE THREAD

**Issue by desert0616** [fchollet/keras#4859]
https://github.com/fchollet/keras/blob/master/keras/engine/training.py#L856

As the mentioned code shows, the validation after each epoch will be firstly separated into batches, and then to collect all the output metrics and reduce them by mean.

I was wondering the meaning of batches of validation, whether it is for the parallel reason? If so, how can I set the batch_sizes of training and validation separately in order to disable the separation of validation?

Actually, the reason why I am struggling with this issue is that I want to use some metrics related to the whole y_true and y_pred, e.g. R-value and normalized MSE. If such metrics are required, should I add a callback on epoch end to obtain the entire matrices and do the particular metrics manually?

Thanks.

**Comment by kudkudak** [comment]
Well, for one in most real-life situations you are not able to pass whole valid at once due to memory restrictions. Also, different batch sizes for valid and train are not common enough probably to warrant separate parameter in API.

Having said that, it is relatively easy to add callback that will compute them for you manually. You can use `LambdaCallback` class and add your metrics to `logs` parameter of callback lambda function.

Alternatively, you could use `fit_generator`, where you can pass generators (and hence have any batch size you wish, even variable one).

**Comment by desert0616** [comment]
Thanks so much for the reply.
